client/app/components/Select.tsx

```diff
@@ -0,0 +1,28 @@
+import React, { useState, useEffect } from "react";
+import { maxBy } from "lodash";
+import AntdSelect, { SelectProps } from "antd/lib/select";
+import { calculateTextWidth } from "@/lib/calculateTextWidth";
+
+function Select({ options, ...props }: SelectProps<any>): JSX.Element {
+  const [dropdownMatchSelectWidth, setDropdownMatchSelectWidth] = useState<number | boolean>(true);
+  useEffect(() => {
+    if (options && options.length > 400) {
+      const largestOpt = maxBy(options, "label.length");
+
+      if (largestOpt) {
+        const offset = 40;
+        const optionText = String(largestOpt.label);
```

Comment:
How about we extend `SelectProps` and `LabeledValue` to enforce the `label` in `LabeledValue` is a `string`? This feels better than casting it here, as this may be unexpected for `Node` values

Reply:
An interesting thing is that since we are using `Lodash` to get the largest value, `null`/`undefined` (actually anything that doesn't have `.label.length`) values would be just "ignored". The enforcement is mostly to be a reminder that it won't work for `non-string` values.